### PR TITLE
Add more common angles into advanceddigitizingdockwidget

### PR DESCRIPTION
**Pull request by lbartoletti**
## Description
This PR adds only some other common angles used in CAD tools, like AutoCAD. 
image

>@3nids I am a bit reluctant at adding so much. It is still easily achievable through the widget by entering the angle (and eventually locking it).

it was a request from colleagues draftsmen, on the contrary I find it almost more useful to have help on angles "less" obvious than orthogonal

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [X] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [X] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [X] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [X] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by 3nids** [comment]
Well, I tend to prefer avoiding adding new configurations while we can avoid them.
So by order of preference, I would:
1. Keep the number of angles as low as possible since you can always achieve this easily in the dock (which was not the case when the common angles were added, angle locking came later)
2. If required add more angles or add an "enter your own" entry (which would be kept as setting and reused)
3. Add configuration for this

**Comment by nyalldawson** [comment]
@3nids is that the final call here? Should this be closed as a "won't fix" pending possible future reimplementations?

**Comment by 3nids** [comment]
I think QGIS should stay as simple as possible and we should avoid add superfluous configuration options (hence my order of preference above). But that's not my decision to take here. 
@lbartoletti what do you think? Is there such a necessity for these angles?

**Comment by lbartoletti** [comment]
Sorry for the delay

1. I don't use the dock angle and the list of angles for the same thing. The first is when I know exactly the angle and I fix it. The second (the object of the PR) is when I have to digitize by snapping successively on common angles.
2. and 3. Isn't it against your feeling that QGIS should stay as simple as possible?

Thanks

**Comment by nyalldawson** [comment]
@lbartoletti 

I'd tend to agree with you - adding options for configuring angles seems needlessly complex

**Comment by nyalldawson** [comment]
Looking at the screenshot at the top of this thread, I really like the approach used there for the menu text. Could we copy that?

**Comment by lbartoletti** [comment]
@nyalldawson replace "Snap to 30° angles" by "30, 60, 90, 120..."?

**Comment by nyalldawson** [comment]
Yes I'd go: "30, 60, 90, 120°..."

**Comment by lbartoletti** [comment]
Aaaah ! Travis is green :)

File: src/gui/qgsadvanceddigitizingdockwidget.cpp [hunk 1]
```diff
@@ -102,13 +108,19 @@ QgsAdvancedDigitizingDockWidget::QgsAdvancedDigitizingDockWidget( QgsMapCanvas *
...
+  mCommonAngleActions = QMap<QAction *, double>();
+  QList< QPair< double, QString > > commonAngles;
+  QString menuText;
+  QList<double> anglesDouble( { 0.0, 5.0, 10.0, 15.0, 18.0, 22.5, 30.0, 45.0, 90.0} );
+  for ( QList<double>::const_iterator it = anglesDouble.constBegin(); it != anglesDouble.constEnd(); ++it )
+  {
+    if ( *it == 0 )
+      menuText = tr( "Do not snap to common angles" );
+    else
+      menuText = QString( tr( "%1, %2, %3, %4°..." ) ).arg( *it, 0, 'f', 1 ).arg( *it * 2, 0, 'f', 1 ).arg( *it * 3, 0, 'f', 1 ).arg( *it * 4, 0, 'f', 1 );
```
Comment (nyalldawson) [inline src/gui/qgsadvanceddigitizingdockwidget.cpp]: '...' -> '…' (proper ellipsis character instead of three dots)

**Comment by nyalldawson** [comment]
@3nids you happy for this to be merged?

**Comment by 3nids** [comment]
yes!

File: src/gui/qgsadvanceddigitizingdockwidget.cpp (hunk 1 above)
Comment (lbartoletti) [reply src/gui/qgsadvanceddigitizingdockwidget.cpp]: Done. Thank you.

**Pull request merged**